src/main/scala/io/iohk/ethereum/network/PeerStatisticsActor.scala

```diff
@@ -0,0 +1,75 @@
+package io.iohk.ethereum.network
+
+import akka.actor._
+import io.iohk.ethereum.network.PeerEventBusActor._
+import io.iohk.ethereum.network.p2p.messages.Codes
+import scala.concurrent.duration.FiniteDuration
+import cats.kernel.Monoid
+
+class PeerStatisticsActor(
+    peerEventBus: ActorRef,
+    var maybeStats: Option[TimeSlotStats[PeerId, PeerStatisticsActor.Stat]]
```

Comment:
Should we add separate settings for it then? I'm maintaining the possibility of not having it because I thought it could be based on the `min-prune-age` which seems like fair play for everyone, old connections and new as well. Otherwise an older peer can have more requests because it had been in the stats longer, and we'll have to resort to taking averages when doing prioritisation. Such settings might be set to zero, which shouldn't mean the system refuses to start. This is why I thought having stats in the `PeerManagerActor` is good because then it's its own jurisdiction of whether it uses them or not. We can potentially extend the interface in a way that `GetStatsFor...` sends a desired duration as well, so different clients can ask for different length windows. Getting complicated though.

Have you given any thought to how you want to use the stats for pruning?

Reply:
In general the whole system will have no idea how long the stats are for, only the programmer knows whether what goes on the `PeerStatisticsActor` makes any sense to what they are doing.

Reply:
Also because someone may decide that we shouldn't waste the time collecting stats at all, and just set `slot-count` to 0, in which case all the other code that rely on it wouldn't have to change, they just have to deal with the possibility of empty stats.

Reply:
Added separate configuration for it and the query messages need to send an explicit window they are interested in.